src/main/java/org/humancellatlas/ingest/schemas/SchemaService.java

```diff
@@ -0,0 +1,84 @@
+    SchemaRepository schemaRepository;
+    private final @NonNull SchemaScraper schemaScraper;
+    private final @NonNull Environment environment;
+
+    public Page<Schema> querySchemas(String highLevelEntity,
+                                     String concreteEntity,
+                                     String domainEntity,
+                                     String subDomainEntity,
+                                     String schemaVersion,
+                                     Pageable pageable){
+        return schemaRepository.findByHighLevelEntityLikeAndConcreteEntityLikeAndDomainEntityLikeAndSubDomainEntityLikeAndSchemaVersionLike(highLevelEntity,
+                                                                                                                                            concreteEntity,
+                                                                                                                                            domainEntity,
+                                                                                                                                            subDomainEntity,
+                                                                                                                                            schemaVersion,
+                                                                                                                                            pageable);
+    }
+
+    @Scheduled(fixedDelay = 1000 * 60 * 60 * 24) // ever 24 hours
+    public void updateSchemasCollection() {
+        schemaScraper.getAllSchemaURIs(URI.create(environment.getProperty("SCHEMA_BASE_URI")))
+                     .forEach(schemaUri -> {
+                         Schema schemaDocument = schemaDescriptionFromSchemaUri(schemaUri);
+
+                         // generate a uuid from the schema namespace
+                         UUID schemaUuid = UUID.nameUUIDFromBytes(schemaUri.toString().getBytes());
+                         schemaDocument.setUuid(new Uuid(schemaUuid.toString()));
+
+                         // delete/update matching schemas
+                         Collection<Schema> matchingSchemas = schemaRepository.findByUuidEquals(new Uuid(schemaUuid.toString()));
+                         schemaRepository.delete(matchingSchemas);
+
+                         schemaRepository.save(schemaDocument);
+                     });
+    }
+
+    public Collection<Schema> schemaDescriptionFromSchemaUris(Collection<URI> schemaUris) {
+        return schemaUris.stream()
+                         .map(this::schemaDescriptionFromSchemaUri)
+                         .collect(Collectors.toList());
+    }
+
+    private Schema schemaDescriptionFromSchemaUri(URI schemaUri) {
+        String[] splitString = schemaUri.toString().split("/");
+        String schemaFullUri = environment.getProperty("SCHEMA_BASE_URI") + schemaUri;
+
+        if(splitString.length == 3) { // then this is a bundle schema
```

Comment:
could add separate tests for each uri "types" (bundle, with subdomain, without subdomain)